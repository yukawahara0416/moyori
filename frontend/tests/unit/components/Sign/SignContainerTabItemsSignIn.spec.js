import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/Sign/SignContainerTabItemsSignIn.vue'

const localVue = createLocalVue()

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Component, {
    localVue,
    stubs: ['policy']
  })
})

describe('getters', () => {
  it('signInFormData', () => {
    expect(wrapper.vm.signInFormData).toEqual(getters.signInFormData())
  })
  it('currentUser', () => {
    expect(wrapper.vm.currentUser).toEqual(getters.currentUser())
  })
  it('headers', () => {
    expect(wrapper.vm.headers).toEqual(getters.headers())
  })
})

describe('v-on', () => {
  it('signInAsTestUser', () => {
    const event = jest.fn()
    wrapper.setMethods({ signInAsTestUser: event })
    wrapper
      .findAll('.v-btn')
      .at(0)
      .trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
  it('signInByTwitter', () => {
    const event = jest.fn()
    wrapper.setMethods({ signIn: event })
    wrapper
      .findAll('.v-btn')
      .at(1)
      .trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
  it('signIn', () => {
    const event = jest.fn()
    wrapper.setMethods({ signIn: event })
    wrapper
      .findAll('.v-btn')
      .at(2)
      .trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('actions', () => {
  it('signIn', () => {
    wrapper.vm.signIn()
    expect(actions.signIn).toHaveBeenCalled()
  })
  // it('signInByTwitter', () => {
  //   wrapper.vm.signIn()
  //   expect(actions.signIn).toHaveBeenCalled()
  // })
  it('signInAsTestUser', () => {
    wrapper.vm.signInAsTestUser()
    expect(actions.signIn).toHaveBeenCalled()
    // expect(actions.signIn).toHaveBeenCalledWith(
    //   expect.any(Object),
    //   getters.signInFormData,
    //   undefined
    // )
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
