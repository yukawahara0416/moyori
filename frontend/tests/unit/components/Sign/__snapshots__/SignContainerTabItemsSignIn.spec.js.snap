// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="container container--fluid"
>
  <div
    class="row align-center justify-center"
    style="flex-direction: column;"
  >
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat success"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              かんたんログイン
            </div>
          </div>
        </header>
         
        <div
          class="v-card__actions"
        >
          <div
            class="row align-center justify-center"
          >
            <div
              class="text-center col"
            >
              <button
                class="v-btn v-btn--contained theme--light v-size--large success"
                type="submit"
              >
                <span
                  class="v-btn__content"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate v-icon--left mdi mdi-ninja theme--light"
                  />
                  
                テストユーザでログイン
              
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat primary"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              SNSアカウントでログイン
            </div>
          </div>
        </header>
         
        <div
          class="v-card__actions"
        >
          <div
            class="row align-center justify-center"
            style="flex-direction: column;"
          >
            <button
              class="my-3 white--text v-btn v-btn--contained theme--light v-size--large"
              style="background-color: rgb(0, 172, 238); border-color: #00acee;"
              type="submit"
            >
              <span
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate v-icon--left mdi mdi-twitter theme--light white--text"
                />
                
              Twitterアカウントでログイン
            
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat primary"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              メールアドレスでログイン
            </div>
          </div>
        </header>
         
        <div
          class="v-card__text"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-email-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-215"
                      style="right: 0px; position: absolute;"
                    >
                      メールアドレス
                    </label>
                    <input
                      id="input-215"
                      name="login"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-lock-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password"
                      style="right: 0px; position: absolute;"
                    >
                      パスワード
                    </label>
                    <input
                      id="password"
                      name="password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="spacer"
          />
           
          <button
            class="mb-3 px-10 v-btn v-btn--contained theme--light v-size--large primary"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
            ログイン
          
            </span>
          </button>
           
          <div
            class="spacer"
          />
        </div>
      </div>
    </div>
     
    <policy-stub />
  </div>
</div>
`;
